library/Garden/Web/RequestInterface.php

```diff
@@ -56,6 +56,21 @@ public function getRoot();
      */
     public function setRoot($root);
 
+    /**
+     * Get the root folder for static files.
+     *
+     * @return string Returns the root as a string.
+     */
+    public function getAssetRoot();
+
+    /**
+     * Set the root folder for static files.
+     *
+     * @param string $root The new root path of files.
+     * @return $this
+     */
+    public function setAssetRoot(string $root);
+
```

Comment:
I can not get the logic why  getAssetRoot and setAssetRoot should belong to this interface?

Reply:
It's there because `->getRoot()` and `setRoot()` are already on that interface. Right now `Gdn_Request/RequestInterface` is where we get information from the server about how to make our URLs. I'd support a proposal for trying to pull a bunch of that stuff out of our request object but at the moment I consider that outside of the scope of the PR.

With the idea that this PR should not require decomposing our Request object (and it cannot without a lot of coordination with infrastructure), than that leaves me with 4 options if I want to write tests for this:

1. Leave the existing `assetRoot` as is and add it to `RequestInterface`. I disagree with this because having a dual-purpose getter/setter does not align with the rest of the methods on `RequestInterface`.
2. The implementation in this PR (deprecate assetRoot in favor of get/set assetRoot).
3. Not write tests involving the assetRoot.
4. Move the asset root setting/getting somewhere else, but still need the `Request` object anyways for the base path.